src/components/Switch.js

```diff
@@ -0,0 +1,54 @@
+import React, {Component} from 'react';
+import {TouchableOpacity, Animated} from 'react-native';
+import PropTypes from 'prop-types';
+import styles from '../styles/styles';
+
+const propTypes = {
+    isOn: PropTypes.bool.isRequired,
+    onToggle: PropTypes.func.isRequired,
+};
+
+class Switch extends Component {
```

Comment:
also I should note I did not try to implement this myself so if there is a reason we can't use a functional/stateless component please let me know since I could very easily have missed something.

Reply:
hmm I can try it out, I think the biggest issue would be that a `new Animated.Value` would be created on every re-render. Normally when using this you throw the value into a ref `useRef(new Animated.Value(0))` but since we're trying to stay away from hooks I ended going with the class implementation.

Reply:
ah okay ya that makes sense. 

so using the class it would actually help reduce needless re-renders as compared to use a functional/stateless component?

Reply:
I guess re-render might not be the problem there but it helps cut down on extra computation on re-rendering?

Reply:
that and I feel like if a re-render were to happen while it's mid animation, it might just jump to the start/end position as well